Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -999,25 +1094,74 @@ bool remove_degenerate_faces(      TriangleMesh& tmesh,
   typedef typename boost::property_traits<VertexPointMap>::value_type Point_3;
   typedef typename boost::property_traits<VertexPointMap>::reference Point_ref;
 
-// First remove edges of length 0
-  bool all_removed = remove_degenerate_edges(edges(tmesh), tmesh, np);
+  std::set<face_descriptor> degenerate_face_set;
+  degenerate_faces(face_range, tmesh, std::inserter(degenerate_face_set, degenerate_face_set.begin()), np);
+
+  if(degenerate_face_set.empty())
+    return true;
 
-  #ifdef CGAL_PMP_REMOVE_DEGENERATE_FACES_DEBUG
+  if(degenerate_face_set.size() == static_cast<std::size_t>(std::distance(faces(tmesh).begin(), faces(tmesh).end())))
+  {
+    clear(tmesh);
+    return true;
+  }
+
+  // Sanitize the face range by adding adjacent degenerate faces
+  std::list<face_descriptor> faces_to_visit(degenerate_face_set.begin(), degenerate_face_set.end());
```

Comment:
I don't really like this part in case we apply it globally, especially for the function is called on the whole mesh. Shall we add a Boolean parameter to avoid the sanitize step in this case?

Reply:
Sure, that's a good idea.